fluent-bit/files/bit.conf.j2

```diff
@@ -64,6 +64,13 @@
 
     {% macro plugin_config(plugins, type) -%}
     {%- for plugin in plugins %}
+      {%-
+        if type == 'OUTPUT'
+        and (plugin.get('Name', '').lower() == 'stackdriver' or plugin.get('name', '').lower() == 'stackdriver')
+        and not pillar['logging']['enabled']
```

Comment:
Want to make sure I follow. The `stackdriver` plugin does not run if google cloud is enabled? The stackdriver plugin is configured for all HLCs in `fluent_bit.sls` but the fluent_bit plugin will ignore it if cloud logging is enabled.

Reply:
You have it flipped. If google cloud logging is not enabled then the stackdriver plugin is ignored. This approach is a bit hacky. Ideally this logic should be present in `fluent_bit.sls` but since that's a pillar file, `pillar['logging']['enabled']` is not accessible.

Reply:
Ahh yeah that would be easier. Just template it based on the logging pillar value there. 

I wish there was a way to load salt-formulas as `states` and not pillars. It makes it confusing when you have to switch between the `fluent-bit` state and the `fluent-bit` formula configuration pillar. 

This hack seems fine. We might want to make it not fail if the pillar doesn't exist. 

Do 
`and (('logging' not in pillar or 'enabled' not in pillar['logging']) and not pillar['logging']['enabled'])

Reply:
I think we should fail if the pillar doesn't exist. It will remind people to specify it in the pillar file. I was initially making it true by default but using defaults can create unnecessary confusion

Reply:
Yeah but it kind of ties the fluent bit formula to our pillar template. Creates a dependency between the two that's hard to see. 

Down the line, if we change the name of `logging` for example, highstates we'll break and we'll have to dive into this logic without a clear reason.

Reply:
True. I actually wanted to move away from the `fluent_bit.sls` file and put everything directly in a config. There is some work involved with that but I will make a TODO here to do that eventually. That way its more robust and there is no hacks needed.

Reply:
Sounds good!